flake.nix

```
@@ -0,0 +1,71 @@
+{
+  description = "An example Golang package";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/nixos-21.11";
+
+  outputs = { self, nixpkgs }:
+    let
+
+      # to work with older version of flakes
+      lastModifiedDate = self.lastModifiedDate or self.lastModified or "19700101";
+
+      # Generate a user-friendly version number.
+      version = builtins.substring 0 8 lastModifiedDate;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
```

We typically alphabetize lists whenever possible for easier readability and to simplify if we ever have to add/remove items from them:
```suggestion
      supportedSystems = [ "aarch64-darwin" "aarch64-linux" "x86_64-darwin" "x86_64-linux" ];
```

---

This conversation appears to have been unintentionally resolved without comment or accepting the suggestion, so I unresolved it.

---

Unless I am misunderstanding and this entire file is part of the auto-generated scaffolding.  If that's the case, then a comment should be added to each of these conversations so that they can be resolved by the original commenter (me).

---

Noted. Yes, this file is generated from a template based on the specific language needed for a flake.